### PR TITLE
fix: Improve Pull Request status reporting

### PR DESCRIPTION
**Pull request by mcdurdin**
Pull Requests don't always show status correctly, for two reasons:

1. GitHub sometimes returns a null status check with a QUEUED state, which the app then saw instead of the correct status check. We see this for the website status checks.

2. We renamed the TC configurations and added versioning into them, which means we need to test only on the parent project name rather than the full status check. This should work into the future as well, when we have new status checks that have the same name pattern.

Also, if any build check fails, this will now be highlighted. The algorithm attempts to find the best corresponding build check for all platforms otherwise, and defaults to the first if no perfect match is found ( which may happen on Common at least).

We still only show one of the status checks in this view but at least we get better reporting of when one fails.

### CODE REVIEW

File: public/src/app/platforms.ts [hunk 1]
```diff
@@ -13,48 +13,48 @@ export const platforms: PlatformSpec[] = [
     id:'android',
     name:'Android',
     configs:{"alpha": "KeymanAndroid_Build", "beta": "KeymanAndroid_Beta", "stable": "KeymanAndroid_Stable", test: "KeymanAndroid_TestPullRequests"},
-    context: "Test (Keyman - Android)"
+    context: "Keyman - Android"
```
Comment (darcywong00) [inline public/src/app/platforms.ts:16]: Do the `configs` match the TC Build configuration IDs?
1. The consolidated alpha/beta/stable builds have `KeymanAndroid_Build`
2. And the Test-13.0 config is keyman_android_test_13_0 (trigger still a TODO_

Comment (mcdurdin) [reply public/src/app/platforms.ts:16]: Yes, the `configs` are supposed to match; I need to do another refactor to support the version-oriented status checks, but for now the changes here support 14.0 build configurations, and I can refactor later.

File: public/src/app/pull-request/pull-request.component.ts [hunk 2]
```diff
@@ -15,27 +15,49 @@ export class PullRequestComponent implements OnInit {
...
+            case 'CANCELLED':
+            case 'SKIPPED':
+            case 'STALE':
+            case 'NEUTRAL':
+            default:
+              if(conclusion == '') conclusion = 'CANCELLED';
+              //return base+'missing'; // various other states
+          }
+        } else {
+          //IN_PROGRESS, QUEUED, REQUESTED
```
Comment (darcywong00) [inline public/src/app/pull-request/pull-request.component.ts:51]: Does `QUEUED` still apply here?

Comment (mcdurdin) [reply public/src/app/pull-request/pull-request.component.ts:51]: l. 30 only applies when there is a `null` status check -- it never starts, never completes, has no associated app, but still appears as an extra status check on some PRs. Looks like a GH bug, perhaps?

**Review by darcywong00 — APPROVED** [review]
lgtm

**Pull request merged**